pandas/io/parsers/python_parser.py

```diff
@@ -205,16 +211,17 @@ class MyDialect(csv.Dialect):
             else:
                 # attempt to sniff the delimiter from the first valid line,
                 # i.e. no comment line and not in skiprows
-                line = f.readline()
+                line: str = f.readline()
```

Comment:
I think you probably don't need this since you have `IO[str]` but it can't hurt.

Reply:
I think this showed an error before, but can try again

Reply:
Yep could remove it